consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala

```diff
@@ -27,8 +27,11 @@ trait BaseApi extends Closable {
     RetryPolicy.backoff(backoffs) {
       // We will assume 5xx are retryable, everything else is not for now
       case (_, Return(rep)) => rep.status.code >= 500 && rep.status.code < 600
+      // Looks like it could be either of 2 depending on where FutureCancelledException occurred
+      case (_, Throw(e: Failure)) if e.isFlagged(Failure.Interrupted) => false
+      case (_, Throw(e: FutureCancelledException)) => false
```

Comment:
From the source code:
```
@deprecated("Provided for API compatibility; use raise() instead.", "6.0.0")
  def cancel() { raise(new FutureCancelledException) }
```

So although `Future.cancel` is deprecated the `FutureCancelledException` is not.

> We should probably return a Failure with Interrupted set -- I think Failure is the new idiomatic way to do this

Can you please explain this in ore details please? How we can influence currently running future this way?